core/src/impl/Kokkos_Utilities.hpp

```diff
@@ -79,6 +79,57 @@ template <class T>
 using remove_cvref_t = typename remove_cvref<T>::type;
 #endif
 
+#if __cplusplus >= 202002
+// C++20 version of std::underlying_type
+//
+// If T is a complete enumeration (enum) type, provides a member typedef type
+// that names the underlying type of T.
+// Otherwise, if T is not an enumeration type, there is no member type.
+// Otherwise (T is an incomplete enumeration type), the program is ill-formed.
+using std::underlying_type;
+using std::underlying_type_t;
+#else
+template <typename E, bool = std::is_enum_v<E>>
+struct underlying_type_impl {};
+
+template <typename E>
+struct underlying_type_impl<E, true> {
+  using type = std::underlying_type_t<E>;
+};
+
+template <typename E>
+using underlying_type = underlying_type_impl<E>;
+
+template <typename E>
+using underlying_type_t = typename underlying_type<E>::type;
+#endif
```

Comment:
It is non-obvious to me why the C++11 definition of `std::underlying_type` would not enable implementing `to_underlying` or `is_scoped_enum`.  Out of curiosity I replaced L82 by `#if 1` and your unit test builds.

Reply:
I think you are correct, but we should be able to rely on the C++20 version, especially if we end up exposing it publicly.